rfcs/text/0011_global_search.md

```diff
@@ -0,0 +1,693 @@
+
+The initial idea for `GlobalSearchResult.url` was to have a polymorphic structure instead of just a string:
+
+```ts
+url: { absUrl: string } | { application: string; path?: string };
+```
+
+That was making it way easier to redirect to an internal result from the UI, as we could directly call
+`application.navigateTo(application, { path })`.
+
+However, that didn't bring answer for the (future) need to be able to search for and redirect to object in 
+different spaces. We could then have changed the structure to
+
+```ts
+url: { absUrl: string } | { application: string; path?: string, space?: string };
+```
+
+But this had some caveats:
+- `space` is an xpack plugin, adding this `space` property in an oss part of the code is problematic
+- The `space` API is not callable from core or oss, meaning that we would have to 'forge' the url to this space anyway
+- this is really not generic. If another plugin was to alter the basepath in another way, we would have needed to add it another property 
+
+So even if the 'parsable absolute url' approach seems fragile, it still felt better than this alternative.
+
+## We could use plain string instead of an enum for `GlobalSearchResult.type`
+
+The current static enum used for type
+
+```ts
+enum SEARCH_TYPE {
+  // non-exhaustive
+  applications = 10,
+  dashboard = 20,
+  visualization = 30,
+  search = 40,
+}
+
+/** @public */
+type GlobalSearchResultType = keyof typeof SEARCH_TYPE;
+
+interface GlobalSearchResult {
+  // [...]
+  type: GlobalSearchResultType;
+}
+```
+
+has some limitations:
+- It forces the enum to be modified every time a new type is added
+- 3rd party plugins cannot introduce new types
+
+We could change the API to accept plain strings for `GlobalSearchResult.type`. However, atm this enum approach 
+is needed as the ordinal values of the entries is used in results sorting. Changing to plain strings forces to find 
+an alternative sorting logic.
+
+## triggered searches could be cancelable
+
+In current specifications, once a search has been triggered, it's not possible to cancel it (see [Search cancellation](#search-cancellation))
+
+Main drawback of this decision is that if a `find` API consumer is going to perform multiple `find` requests and 
+only uses the last one (For example in the UI, a debounced `find` triggered on end-user keypress: On every new keypress,
+a new search is performed and the previous, potentially still executing, `find` call will still be executed), it can't
+cancel the previous active search, resulting on potential useless http calls and resources consumption.
+
+We could add an optional cancellation signal or observable in `GlobalSearchOptions` and propagate it to result providers.
+That way `find` calls could be aborted.
+
+However result providers would still have to manually handles this signal to cancel any http call or 
+other asynchronous task that could be pending. 
```

Comment:
This feels rather easy to implement. Maybe it should be in the initial implementation?

Reply:
If it's easy to implement, I'm not opposed to us doing so with the initial implementation.

Reply:
added in https://github.com/elastic/kibana/pull/64284/commits/3a094aae6bf8b96100ed1e77d64314ca1e4ef646